### ISSUE TITLE
Brightness Controller Gui Appears But Does Not Seem To Affect Output

### ISSUE THREAD

**Issue by sarmstr5** [LordAmit/Brightness#76]
I didnt know exactly what information you would find helpful so I just tried to get everything seemingly relevant. 
 - Running Gnome Ubuntu 17.04. 
 - GPU- AMD Fury. 
 - I have three monitors.  

Gui appears and seems to function but does not affect my display.  I turned off Gnome's Night Light feature from the 'Displays' setting.  Based on the output below (attached the whole report) I thought it would be helpful to see what happens when I ran 'xrandr' in the terminal.  The xrandr output is below is the setupt output and also attached.  Seems like your code 'detect_display_devices()' should be grabbing DisplayPort2 but the words list is empty.  I am not sure why 'connected' is not matching.  I haven't actually tested the code, just my guess.  Please let me know if there is anything else I need/can report.
[xrandr_output.txt](https://github.com/LordAmit/Brightness/files/966541/xrandr_output.txt)
[brighness-master_output.txt](https://github.com/LordAmit/Brightness/files/966540/brighness-master_output.txt)
![screenshot from 2017-04-29 23-58-33](https://cloud.githubusercontent.com/assets/10838957/25561340/ee5b531a-2d37-11e7-9180-595627c99662.png)

Respectfully,
Shane

----------------initial terminal output after setup----------
Downloads/temp_for_zip/Brightness-master$ python src/init.py
warning: output None not found; ignoring
xrandr: Need crtc to set gamma on.
Traceback (most recent call last):
  File "src/init.py", line 146, in change_value_pbr
    Executor.execute_command(cmd_value)
  File "/home/shane/Downloads/temp_for_zip/Brightness-master/src/util/executor.py", line 23, in execute_command
    subprocess.check_output(string_cmd, shell = True)
  File "/usr/lib/python2.7/subprocess.py", line 219, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command 'xrandr        --output None         --brightness 0.99        --gamma 1.0:1.0:1.0' returned non-zero exit status 1
warning: output None not found; ignoring
xrandr: Need crtc to set gamma on.
Traceback (most recent call last):
  File "src/init.py", line 146, in change_value_pbr
    Executor.execute_command(cmd_value)
  File "/home/shane/Downloads/temp_for_zip/Brightness-master/src/util/executor.py", line 23, in execute_command
.... Continues for 1.5k lines ------------------

-------------xrandr output-----------
Screen 0: minimum 320 x 200, current 7200 x 2160, maximum 16384 x 16384
DisplayPort-0 disconnected (normal left inverted right x axis y axis)
DisplayPort-1 disconnected (normal left inverted right x axis y axis)
DisplayPort-2 connected primary 3840x2160+1440+0 (normal left inverted right x axis y axis) 600mm x 340mm
   3840x2160     60.00*+  30.00  
   2560x1440     59.95  
   1920x1080     60.00    59.94  
   1600x900      60.00  
   1280x1024     60.02  
   1280x800      59.81  
   1152x864      59.97  
   1280x720      60.00    59.94  
   1024x768      60.00  
   800x600       60.32  
   720x480       60.00    59.94  
   640x480       60.00    59.94  
HDMI-A-0 connected 1920x1080+5280+540 (normal left inverted right x axis y axis) 521mm x 293mm
   1920x1080     60.00*+  50.00    59.94  
   1920x1080i    60.00    50.00    59.94  
   1680x1050     59.88  
   1280x1024     75.02    60.02  
   1440x900      59.90  
   1280x960      60.00  
   1280x800      59.91  
   1152x864      75.00  
   1280x720      60.00    50.00    59.94  
   1440x576      50.00  
   1024x768      75.03    70.07    60.00  
   1440x480      60.00    59.94  
   832x624       74.55  
   800x600       72.19    75.00    60.32    56.25  
   720x576       50.00  
   720x480       60.00    59.94  
   640x480       75.00    66.67    60.00    59.94  
   720x400       70.08  
DVI-D-0 connected 1440x900+0+882 (normal left inverted right x axis y axis) 410mm x 256mm
   1440x900      59.89*+  74.98  
   1280x960      60.00  
   1152x864      75.00  
   1024x768      75.03    70.07    60.00  
   832x624       74.55  
   800x600       72.19    75.00    60.32    56.25  
   640x480       75.00    72.81    66.67    59.94  
   720x400       70.08

**Comment by LordAmit** [comment]
Hi, This is a bit weird. I have tested your provided file with the following: 
```def detect_display_devices():
    """
    Detects available displays.
    returns connected_devs
    This contains the available device names compatible with xrandr
    """
    connected_devs = []

    xrandr_output = open("xrandr_output.txt", "r").read()
    lines = xrandr_output.split('\n')
    for line in lines:
        words = line.split(' ')
        for word in words:
            if word == 'connected':
                connected_devs.append(words[0])
    return connected_devs


if __name__ == "__main__":
    print(detect_display_devices())
```

The output is:
```['DisplayPort-2', 'HDMI-A-0', 'DVI-D-0']```
It seems like it is catching the display. Not sure why it is forwarding None to you.


Can you please download this file, rename it to test_check_displays.py, run and then provide the output? Will really appreciate it.

[test_check_displays.txt](https://github.com/LordAmit/Brightness/files/967544/test_check_displays.txt)

**Comment by sarmstr5** [comment]
Will do later tonight.  Thanks!

**Comment by sarmstr5** [comment]
Hey just wanted to clarify for later.  
- You dont care which directory I run this in.  
- You want me to rerun xrandr with no additional parameters.
- Feed that output into the program
- Provide you the output (which should look like your output)
- As an aside, if they dont line up I will look a little bit more into the code and send you the printout from 'lines' as well.
Thanks for the help.

**Comment by LordAmit** [comment]
Ah, I should have explained further. Sorry for the confusion. 

It does not matter which directory you are in.  The file I provided is a standalone file, which will be used to see if the current program is working properly. 

You have to run it from terminal, like this:

```python test_check_displays.py```

It will provide output on your screen. if you want to save it, you can do this: 

```python test_check_displays.py > xrandr_output_check.txt```

And simply upload the file here.

First, I will see if the display detecting part is working properly there. If it is detecting properly, the problem is elsewhere.

**Comment by sarmstr5** [comment]
Hey I ran the code and that wasnt the issue.  
As a heads up the GUI comes up fine without any issues but once i try to change the settings it runs into errors.

[xrandr_output_check.txt](https://github.com/LordAmit/Brightness/files/969017/xrandr_output_check.txt)

**Comment by LordAmit** [comment]
Hi,
Can you please download the file, rename it to init.py, and then replace the current init.py file? 
[init.txt](https://github.com/LordAmit/Brightness/files/970789/init.txt)

The change is made in the constructor method, where it assigns the display names to display1 and display2.  I am merely checking for the values.
```python
def __init__(self, parent=None):
        """Initializes"""
        QtGui.QMainWindow.__init__(self, parent)

        self.ui = Ui_MainWindow()
        self.ui.setupUi(self)

        self.display1 = None
        self.display2 = None
        self.no_of_connected_dev = 0
        self.__assign_displays()
        >>> print self.display1
        >>> print self.display2
        self.values = []
        self.array_value = 0.01
        for i in xrange(0, 100):
            self.values.append(self.array_value)
            self.array_value += 0.01

        self.connect_handlers()

        self.setup_widgets()
```
If you simply run it from terminal, it will print the values received from the `check_displays()` method. 

Really sorry for the trouble. :disappointed:

**Comment by LordAmit** [comment]
@sarmstr5  - any update?

**Comment by sarmstr5** [comment]
Hey sorry in grad school and I have midterms this week, I spent some time diagnosing the issue on friday and the script worked for me (got the correct output).  The problem though is I accidentally installed pyside in a conda environment that I was using earlier (that was using a different version of python).  So I think the issue has to do with the fact that I dont have pyside installed outside of that conda environment.  I am running python 3.6 and pyside is not compatible with that version.  I tried using PIP, sudo apt install, and conda install but no go.  I need to do research on if i can install it (i am not sure why it would be compatible with 3.4 but not 3.6).  If i cant i will look into if there is a way to run it with a conda environment or if I can force it to run with python 2.7.  I was working on this friday so i might have messed up some nuisance.  Let me know if you have any ideas on how I should go about install it and sorry about the delayed response.

R/
Shane
